core/src/main/scala/org/http4s/parser/QueryParser.scala

```diff
@@ -110,7 +110,7 @@ private[http4s] object QueryParser {
   private case object KEY extends State
   private case object VALUE extends State
 
-  private val QChars = BitSet((Pchar ++ "/?".toSet - '&' - '=').map(_.toInt).toSeq:_*)
+  private val QChars = BitSet((Pchar ++ "/?[]".toSet - '&' - '=').map(_.toInt).toSeq:_*)
```

Comment:
We have tried to format parsers like the specification, if possible. ```QChars``` represents ```query``` (probably would have been better as ```Query``` but that name is taken by a type) is defined [here](https://tools.ietf.org/html/rfc3986#section-3.4).

Could we make a new val, maybe ```ExtendedQChars``` that adds the expanded chars to ```QChar``` with a comment on its origin and why its inconsistent with the spec. Then I think we can make the class ```QueryParser``` take the valid chars ```BitSet``` as a parameter with a default of ```ExtendedQChars```.